src/System.Memory/src/System/SpanHelpers.cs

```diff
@@ -12,6 +12,76 @@ namespace System
     internal static partial class SpanHelpers
     {
         /// <summary>
+        /// Implements the core copy functionality used by Span and ReadOnlySpan.
+        ///
+        /// NOTE: Fast span implements TryCopyTo in core and therefore this implementation
+        ///       is only used by portable span. The code must live in code that only compiles
+        ///       for portable span which means either each individual span implementation
+        ///       of this shared code file. Other shared SpanHelper.X.cs files are compiled
+        ///       for both portable and fast span implementations.
+        /// </summary>
+        public static unsafe void CopyCore<T>(ref T dst, int dstLength, ref T src, int srcLength)
```

Comment:
Can we name it CopyTo instead of CopyCore? I understand that this is the workhorse for the CopyTo functionality, but other Helper methods have the same name as the method they "help". Also "core" can refer to corelib (as seen in the xml comments for this method).